Source/Serialization.Protobuf/Serializer.cs

```diff
@@ -167,15 +167,11 @@ object ReadValue(CodedInputStream inputStream, object value, Type type, Type tar
             {
                 value = inputStream.ReadBool();
             }
-            else if (type == typeof(DateTimeOffset))
+            else if (type == typeof(DateTimeOffset) || type == typeof(DateTime))
             {
-                value = DateTimeOffset.FromFileTime(inputStream.ReadInt64());
+                value = DateTimeOffset.FromUnixTimeMilliseconds(inputStream.ReadInt64());
+                if (type == typeof(DateTime)) value = ((DateTimeOffset)value).UtcDateTime;
```

Comment:
I like this.  Much easier to go through DateTimeOffset than deal with DateTimeKind